beacon-chain/blockchain/receive_block.go

```diff
@@ -45,6 +46,9 @@ func (s *Service) ReceiveBlock(ctx context.Context, block interfaces.ReadOnlySig
 		return err
 	}
 
+	s.cfg.ForkChoiceStore.Lock()
+	defer s.cfg.ForkChoiceStore.Unlock()
```

Comment:
`s.cfg.ForkChoiceStore.IsOptimistic` is also called within `onBlock`

Reply:
Same with finalized and justified epochs

Reply:
this is not a problem, the problem is calls to `chaininfo.IsOptimistic` any access to forkchoice directly is not a problem